src/matlab2tikz.m

```diff
@@ -4825,8 +4825,48 @@ function warnFacetedInterp(m2t)
             cbarStyleOptions = opts_add(cbarStyleOptions,...
                 'xticklabel pos', 'upper');
         case 'southoutside'
-
             cbarTemplate = 'horizontal';
+        case 'manual'
+            origUnits = handle.Units;
+            origAxesUnits = handle.Axes.Units;
+            handle.Units = 'centimeters';        % We need absolute
+            handle.Axes.Units = 'centimeters';   % positions for tikz
+            cbarPos = handle.Position;
+            cbarAxesPos = handle.Axes.Position;
+            handle.Units = origUnits;
+            handle.Axes.Units = origAxesUnits;
+
+            % Cases of colorbar axis locations (in or out) depending on position
+            % of colorbar relative to it's associated axes.
+            % According to matlab manual (R2016a) colorbars with Location 'manual'
+            % can only be vertical.
+            if cbarPos(1) + 0.5*cbarPos(3) < cbarAxesPos(1) + 0.5*cbarAxesPos(3) 
```

Comment:
If would write this calculation using intermediate variables:

```matlab
center = @(dims) (dims.left + dims.right)/2;
centerCbar = center(pos2dims(cbarPos));
centerAxes = center(pos2dims(cbarAxesPos));
```

to make it more readable.